services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/dataset_validator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Research
+    module GenAI
+      class DatasetValidator < ApplicationService
+        attr_reader :file
+
+        OPTIMAL = HasAssignedStatus::OPTIMAL
+        SUBOPTIMAL = HasAssignedStatus::SUBOPTIMAL
+        MISSING_FEEDBACK_ERROR = 'is missing feedback. Please add Curriculum Proposed Feedback or AutoML Primary Feedback.'
+
+        def initialize(file:)
+          @file = file
+        end
+
+        def run
+          [].tap do |errors|
+            CSV.parse(file.read, headers: true).each.with_index(2) do |row, index|
+              next if row['Data Partition'] == 'test'
```

Comment:
We might consider having the Curriculum Team upload a `csv` without headers and go by the column. This seems a little fragile that the headers need to be exact matches.